src/reducers/screenReducers/posts.js

```diff
@@ -0,0 +1,48 @@
+/* @flow */
+import {
+  ADD_POST,
+  DELETE_POST,
+  UPDATE_POST,
+  GET_POSTS
+} from "../../constants/actionTypes";
+import type {
+  Post,
+  AddPostAction,
+  DeletePostAction,
+  UpdatePostAction,
+  GetPostsAction
+} from "../../types/types";
+
+type State = {
+  +posts: Array<Post>
+};
+
+export type Action =
+  | AddPostAction
+  | DeletePostAction
+  | UpdatePostAction
+  | GetPostsAction;
+
+const postsState = { posts: [] };
+
+export default function postsReducer(
+  state: State = postsState,
+  action: Action
+): State {
+  switch (action.type) {
```

Comment:
Actually, I realised that this reducer is great to be as Array instead of Object, so you can move the logic back to this reducer, but follow this example https://github.com/reduxjs/redux/blob/master/examples/todomvc/src/reducers/todos.js